src/wp-includes/global-styles-and-settings.php

```diff
@@ -285,13 +285,8 @@ function wp_theme_has_theme_json() {
 	}
 
 	// Does the theme have its own theme.json?
-	$theme_has_support = is_readable( get_stylesheet_directory() . '/theme.json' );
-
-	// Look up the parent if the child does not have a theme.json.
-	if ( ! $theme_has_support ) {
-		$theme_has_support = is_readable( get_template_directory() . '/theme.json' );
-	}
-
+	$theme_has_support = is_readable( get_theme_file_path( 'theme.json' ) );
```

Comment:
Noting there's a filter `'theme_file_path'` in this function, meaning the returned string passes through a filter. That's good and also potentially problematic.

Why problematic? 

What if the filter changes the returned string to a different file other than `theme.json`? If that file exists, then this function will return `true` to indicate that yes the theme does support `theme.json`.

Yes, that's the developer's responsibility to ensure their callback logic works correctly.

But I'm also thinking that in this specific function, the risk is to returning and static variable caching the wrong state, making it harder for the developer to debug.

I'm thinking to keep the logic as it is. 

@felixarntz @spacedmonkey @oandregal What do you think?

Reply:
@hellofromtonya I don't really see this as a concern, as the function is intended for this purpose and is called in many different places in core. If someone _wants_ to break WordPress using a filter in the wrong way, they can in 1000s of places.

At the same time, there is one other (very minor, but potentially relevant) difference in the logic: The code here uses `is_readable` while `get_theme_file_path()` uses `file_exists`. FWIW, I do not understand why the `theme.json` code uses `is_readable` in general, as it has been common in Core to simply rely on `file_exists`, and typically there shouldn't be any difference. Do you have any context on that @oandregal?

Reply:
!00% agree with @felixarntz . You can break wordpress with filters, everyone knows that, it is not a reason not to use correctly use this function here.

Reply:
Thanks for raising, Tonya. I'd think this change should be fine: the consumers of the filter are responsible for making sure it works, and the cache is non-persistent (the value is calculated in every request).

> FWIW, I do not understand why the theme.json code uses is_readable in general, as it has been common in Core to simply rely on file_exists, and typically there shouldn't be any difference. Do you have any context on that @oandregal?

I see plenty of uses for both [is_readable](https://github.com/WordPress/wordpress-develop/search?q=is_readable) and [file_exists](https://github.com/WordPress/wordpress-develop/search?q=file_exists) in core.

In this case, we want to read the file later, so we check for `is_readable`. Note that this function and the function that actually reads the file from the file system use the same check `is_readable` [check](https://github.com/WordPress/wordpress-develop/blob/5cf1b5ac22c58e4492ae68bac60715213da3181f/src/wp-includes/class-wp-theme-json-resolver.php#L621). If there was a reason to update this check, it'll need to be updated in both places.

Reply:
I agree with @oandregal here. Let's stick with `is_readable`.